changelog/fix19785.dd

```diff
@@ -0,0 +1,44 @@
+Function Parameter Name Mangling Changes
+
+If the parameter type can be implicitly converted to a type
+that at the top level does not have `const` or `shared` or `inout` or `immutable`
+then the converted type is mangled instead. This means
+these functions mangle to the same string:
+
+---
+void foo(immutable int i);
+void foo(const int i);
+void foo(int i);
+
+void foo(const int* p);
+void foo(const(int)* p);
+---
+
+Note that C++ compilers typically do this as well.
```

Comment:
Not on Windows, e.g. `void foo(const int* const ptr)` is mangled as `?foo@@YAXQBH@Z` by both  cl and dmc. This translates back to `void __cdecl foo(int const * const)`.

The change in this PR might help interfacing with `void foo(const Class* ptr)`, though.

Reply:
DMC++ tries to duplicate what VC++ does for mangling. Where it would show up in semantics is how the two functions overload against each other. Obviously, if they mangle the same they have to be semantically regarded has having the same function ABI and cannot overload against each other.